Cabal/src/Distribution/Simple/Utils.hs

```diff
@@ -372,7 +379,61 @@ die' verbosity msg = withFrozenCallStack $ do
     =<< annotateErrorString verbosity
     =<< pure . wrapTextVerbosity verbosity
     =<< pure . addErrorPrefix
-    =<< prefixWithProgName msg
+    =<< prefixWithProgName msg      
+
+-- Type which will be a wrapper for cabal -expections and cabal-install exceptions
+data VerboseException a  = VerboseException CallStack POSIXTime Verbosity a
+ deriving (Show, Typeable)
+
+-- A new dieWithException function which will replace the existing die' call sites
+dieWithException :: HasCallStack => Verbosity -> CabalException -> IO a
```

Comment:
All of these cases can be reduced away to just wrapping things.

Reply:
Incorporated the comment.